internal/services/sources_service.go

```diff
@@ -47,3 +49,44 @@ func GetSource(w http.ResponseWriter, r *http.Request) {
 		renderError(w, r, payloads.NewRenderError(r.Context(), "show source", err))
 	}
 }
+
+func fetchARN(ctx context.Context, client sources.SourcesIntegration, sourceId string) (string, error) {
+	// Get all the authentications linked to a specific source
+	resp, err := client.ListSourceAuthenticationsWithResponse(ctx, sourceId, &sources.ListSourceAuthenticationsParams{}, AddIdentityHeader)
+	if err != nil {
+		return "", fmt.Errorf("cannot list source authentication: %w", err)
+	}
+
+	// Filter authentications to include only auth where resource_type == "Application"
+	auth, err := filterSourceAuthentications(resp.JSON200.Data)
+	if err != nil {
+		return "", err
+	}
+	// Get the resource_id which equals to application_id
+	// and check that application_type_id in /applications/<app_id> equals to provisioning id
+	res, err := client.ShowApplicationWithResponse(ctx, *auth.ResourceId, AddIdentityHeader)
+	if err != nil {
+		return "", fmt.Errorf("cannot list source authentication: %w", err)
+	}
+
+	if *res.JSON200.ApplicationTypeId == config.Sources.AppTypeId {
+		return *auth.Username, nil
+
+	}
+	return "", fmt.Errorf("cannot find authentication linked to source id %s and to the provisioning app: %w", sourceId, err)
+}
+
+func filterSourceAuthentications(authentications *[]sources.AuthenticationRead) (sources.AuthenticationRead, error) {
+	auths := *authentications
+	list := make([]sources.AuthenticationRead, 0, len(auths))
+	for _, auth := range auths {
+		if *auth.ResourceType == "Application" {
+			list = append(list, auth)
+		}
+	}
+	// Assumption: each source has one authentication linked to it
+	if len(list) > 1 {
+		return sources.AuthenticationRead{}, sources.MoreThenOneAuthenticationForSourceErr
```

Comment:
Is this correct in case of superkey Source? The superkey source can have multiple authentications (for multiple apps) IMHO.

Reply:
Good point, I think It will be better to get all the auths and search for the one with the correct resource Id. 
I'll fix it tomorrow.